compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7024,6 +7031,27 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, List<String> includedRecordFields) {
+        if (Symbols.isFlagOn(param.flags, Flags.INCLUDED)) {
+            Set<String> fields = ((BRecordType) Types.getReferredType(param.type)).fields.keySet();
+            includedRecordFields.addAll(fields);
+        }
+    }
+
+    // If there is a named-arg or positional-arg corresponding to an included-record-param,
+    // it is an error for a named-arg to specify a field of that included-record-param.
+    private void checkSameNamedArgsInIncludedRecords(List<BLangExpression> namedArgs, List<String> incRecordFields,
+                                                     List<String> includedRecordParamNames) {
+        incRecordFields.removeIf(field -> !includedRecordParamNames.contains(field));
+        for (BLangExpression namedArg : namedArgs) {
+            String argName = ((NamedArgNode) namedArg).getName().value;
+            if (incRecordFields.contains(argName)) {
```

Comment:
Since we do searching here I believe a hash set is the more accurate data structure here. Searching on array list is O(n) where we are doing lots of searching here inside another loop making it O(n<sup>2</sup>).

Reply:
Changed, thank you for the comment